proto/bootz.proto

```diff
@@ -0,0 +1,167 @@
+// control-card modules.
+message ChassisDescriptor {
+  string manufacturer = 1;
+  string part_number = 2;
+  // This field is populated for fixed form-factor chassis, and empty for
+  // modular chassis.
+  string serial_number = 3;
+  // For modular chassis this field will be used to list the control cards
+  // to be bootstrapped.
+  repeated ControlCard control_cards = 4;
+}
+
+// Details of the control card, including serial-number and the location
+// of the card on the chassis.
+message ControlCard {
+  string part_number = 1;
+  string serial_number = 2;
+  int32 slot = 3;
+}
+
+// The status of each control card (i.e. initialized or not). For first install
+// of the chassis, both control cards are in NOT_INSTALLED state. For 
+// install a hot-swapped standby, the status of the active card is INSTALLED, 
+// and that of the standby is NOT_INSTALLED. For FF chassis, the serial number
+// is that of the chassis, and status is NOT_INSTALLED.
+message ControlCardState {
+  string serial_number = 1;
+  ControlCardStatus status = 2;
+}
+
+enum ControlCardStatus {
+  CONTROL_CARD_STATUS_UNSPECIFIED = 0;
+  CONTROL_CARD_STATUS_NOT_INSTALLED = 1;
+  CONTROL_CARD_STATUS_INSTALLED = 2;
+}
+
+message GetBootstrapDataResponse {
+  message BootstrapDataResponse {
+    // The serial number of the control card to which this state should
+    // be applied.
+    string serial_num = 1;
+    // The device should download and install this image (or skip if the
+    // device is already running it).
+    SoftwareImage intended_image = 2;
+    // The device should set this hash as its Bootloader password.
+    string boot_password_hash = 3;
+    // This certificate should be used to validate the server when reporting
+    // progress.
+    string server_trust_cert = 4;
+    // Boot configuration is specified as structured data.
+    BootConfig boot_config = 5;
+    Credentials credentials = 6;
+    gnsi.pathz.v1.UploadRequest pathz = 7;
+    gnsi.authz.v1.UploadRequest authz = 8;
+    gnsi.certz.v1.UploadRequest certificates = 9;
+  } 
+  repeated BootstrapDataResponse responses = 1;
+  bytes ownership_voucher = 101;
+  bytes ownership_certificate = 102;
+  // This is a signature generated over the bytes of the response_body field,
+  // using the ownership_certificate.
+  string response_signature = 103;
+}
+
+message SoftwareImage {
+  string name = 1;
+  string version = 2;
+  // An HTTP location from where the OS file can be downloaded.
+  string url = 3;
+  // Hash of the OS image. The device must verify that the hash of the
+  // downloaded OS image matches this.
+  string os_image_hash = 4;
+  string hash_algorithm = 5;
+}
+
+message Credentials {
+  repeated gnsi.credentialz.v1.AuthorizedKeysRequest credentials = 1;
+  repeated gnsi.credentialz.v1.AuthorizedUsersRequest users = 2;
+  repeated gnsi.credentialz.v1.PasswordRequest passwords = 3;
+}
+
+message BootConfig {
+  string hostname = 1;
```

Comment:
wouldn't hostname just be part of the *_config"?  Or maybe there is a use case for hostname to be outside of the config?